src/Shared/src/CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -26,10 +26,13 @@ internal class CommandLineApplication
         // options.
         private readonly bool _continueAfterUnexpectedArg;
 
-        public CommandLineApplication(bool throwOnUnexpectedArg = true, bool continueAfterUnexpectedArg = false)
+        private readonly bool _treatUnmatchedOptionsAsArguments;
+
+        public CommandLineApplication(bool throwOnUnexpectedArg = true, bool continueAfterUnexpectedArg = false, bool treatUnmatchedOptionsAsArguments = false)
```

Comment:
@anurse  Does this count as a breaking change? I don't know if I pick the best option on how to pass `treatUnmatchedOptionsAsArguments`. On the other hand I don't think many users are using `CommandLineApplication`.

Reply:
No, this component is shared-source and not published to NuGet. There's a third-party (though it was written by a now-former team member) [fork of it](https://github.com/natemcmaster/CommandLineUtils), but this code shouldn't be visible outside of our repos and that fork has long-since drifted away from this source.